### PR TITLE
Update RPC documentation for commands involving subaddresses

### PR DESCRIPTION
**Pull request by coneiric**
References #694.

Includes documentation and examples for new commands dealing with subaddresses.
Updates for existing commands that now deal with subaddresses. Some fields unrelated to subaddresses were missing, and are now included.

### CODE REVIEW

File: _i18n/template/resources/developer-guides/wallet-rpc.md [hunk 1]
```diff
@@ -78,34 +87,56 @@ Note: "atomic units" refer to the smallest fraction of 1 XMR according to the mo
...
-Inputs: *None*.
+Inputs:
+
+* *account_index* - unsigned int; Return balance for this account.
 
 Outputs:
 
 * *balance* - unsigned int; The total balance of the current monero-wallet-rpc in session.
 * *unlocked_balance* - unsigned int; Unlocked funds are those funds that are sufficiently deep enough in the Monero blockchain to be considered safe to spend.
+* *multisig_import_needed* - boolean; True if balance transferred to a multisig wallet.
```
Comment (moneromooo-monero) [inline _i18n/template/resources/developer-guides/wallet-rpc.md]: Incorrect. It's true if importing multisig data from other signers is needed to get correct balance

File: _i18n/template/resources/developer-guides/wallet-rpc.md [hunk 2]
```diff
@@ -78,34 +87,56 @@ Note: "atomic units" refer to the smallest fraction of 1 XMR according to the mo
...
+* *per_subaddress* - array of subaddress information; Balance information for each subaddress in an account.
+  * *address_index* - unsigned int; Index of the subaddress in the account.
+  * *address* - string; Base64 representation of the subaddress.
```
Comment (moneromooo-monero) [inline _i18n/template/resources/developer-guides/wallet-rpc.md]: Incorrect. It is the address (which is base58 of the public keys, mostly).

File: _i18n/template/resources/developer-guides/wallet-rpc.md [hunk 3]
```diff
@@ -78,34 +87,56 @@ Note: "atomic units" refer to the smallest fraction of 1 XMR according to the mo
...
 ### **getaddress**
 
-Return the wallet's address.
+Return the wallet's addresses for an account. Optionally filter for specific set of subaddresses.
 
-Inputs: *None*.
+Inputs:
+
+* *account_index* - unsigned int; Return subaddresses for this account.
+* *address_index* - array of unsigned int; List of subaddresses to return from an account.
```
Comment (moneromooo-monero) [inline _i18n/template/resources/developer-guides/wallet-rpc.md]: optional

File: _i18n/template/resources/developer-guides/wallet-rpc.md [hunk 4]
```diff
@@ -114,13 +145,24 @@ Outputs:
 Example:
 
 ```
-$ curl -X POST http://127.0.0.1:18082/json_rpc -d '{"jsonrpc":"2.0","id":"0","method":"getaddress"}' -H 'Content-Type: application/json'
+$ curl -X POST http://127.0.0.1:18082/json_rpc -d '{"jsonrpc":"2.0","id":"0","method":"getaddress","params":{"account_index":'$ACCT',"address_index":[0,1]}}' -H 'Content-Type: application/json'
```
Comment (moneromooo-monero) [inline _i18n/template/resources/developer-guides/wallet-rpc.md]: Using a variable hides the format, not very useful for an example

File: _i18n/template/resources/developer-guides/wallet-rpc.md [hunk 5]
```diff
@@ -168,15 +212,19 @@ Inputs:
 * *priority* - unsigned int; Set a priority for the transaction. Accepted Values are: 0-3 for: default, unimportant, normal, elevated, priority.
 * *do_not_relay* - boolean; (Optional) If true, the newly created transaction will not be relayed to the monero network. (Defaults to false)
 * *get_tx_hex* - boolean; Return the transaction as hex string after sending
+* *get_tx_metadata* - boolean; Return the transaction metadata.
```
Comment (moneromooo-monero) [inline _i18n/template/resources/developer-guides/wallet-rpc.md]: Kinda tautoligical. Used to relay this transaction later on.

File: _i18n/template/resources/developer-guides/wallet-rpc.md [hunk 6]
```diff
@@ -168,15 +212,19 @@ Inputs:
...
 
 * *fee* - Integer value of the fee charged for the txn.
 * *tx_hash* - String for the publically searchable transaction hash
 * *tx_key* - String for the transaction key if get_tx_key is true, otherwise, blank string.
+* *amount_keys* - List of strings for the amount keys.
+* *amount* - Amount transferred for the transaction.
 * *tx_blob* - Transaction as hex string if get_tx_hex is true
-
+* *tx_metadata* - Transaction metadata string if get_tx_metadata is true.
+* *multisig_txset* - Set of signing keys used in a multisig transaction (empty for non-multisig).
```
Comment (moneromooo-monero) [inline _i18n/template/resources/developer-guides/wallet-rpc.md]: Incorrect, it is a set of transactions in the process of being signed.

File: _i18n/template/resources/developer-guides/wallet-rpc.md [hunk 7]
```diff
@@ -204,15 +257,18 @@ Inputs:
 * *destinations* - array of destinations to receive XMR:
   * *amount* - unsigned int; Amount to send to each destination, in atomic units.
   * *address* - string; Destination public address.
-* *fee* - unsigned int; Ignored, will be automatically calculated.
+* *account_index* - unsigned int; Transfer from this account.
+* *subaddr_indices* - array of unsigned int; Transfer from this set of subaddresses.
 * *mixin* - unsigned int; Number of outpouts from the blockchain to mix with (0 means no mixing).
+* *ring_size* - unsigned int; Amount to adjust mixin. Accepted values: 0 (default mixin, currently 6), n >= default (set mixin to n).
```
Comment (moneromooo-monero) [inline _i18n/template/resources/developer-guides/wallet-rpc.md]: Incorrect, it sets ring size to n (mixin+1)

File: _i18n/template/resources/developer-guides/wallet-rpc.md [hunk 8]
```diff
@@ -271,20 +329,25 @@ Send all unlocked balance to an address.
...
 * *get_tx_hex* - boolean; (Optional) return the transactions as hex encoded string.
+* *get_tx_metadata* - boolean; (Optional) return the transaction metadata as a string.
 
 Outputs:
 
 * *tx_hash_list* - array of string;
 * *tx_key_list* - array of string;
 * *tx_blob_list* - array of string; 
+* *tx_metadata_list* - array of string;
+* *tx_multisig_set* - string;
```
Comment (moneromooo-monero) [inline _i18n/template/resources/developer-guides/wallet-rpc.md]: Missing description (plus plenty below)

File: _i18n/template/resources/developer-guides/wallet-rpc.md [hunk 9]
```diff
@@ -507,6 +611,9 @@ Return a list of incoming transfers to the wallet.
 Inputs:
 
 * *transfer_type* - string; "all": all the transfers, "available": only transfers which are not yet spent, OR "unavailable": only transfers which are already spent.
+* *account_index* - unsigned int; Return transfers for this account.
+* *subaddr_indices* - array of unsigned int; Return transfers sent to these subaddresses.
+* *verbose* - boolean; Enable verbose output, return key image if true.
```
Comment (moneromooo-monero) [inline _i18n/template/resources/developer-guides/wallet-rpc.md]: What a weird parameter.Documentation is correct, but I'll probably change it to make more sense.

File: _i18n/template/resources/developer-guides/wallet-rpc.md [hunk 10]
```diff
@@ -1214,3 +1335,262 @@ $ curl -X POST http://localhost:18082/json_rpc -d '{"jsonrpc":"2.0","id":"0","me
...
+Outputs:
+
+* *subaddress_accounts* - array of subaddress account information;
+  * *account_index* - unsigned int; Index of the account.
+  * *balance* - unsigned int; Balance of the account (locked or unlocked).
+  * *base_address* - string; Base64 representation of the first subaddress in the account.
+  * *label* - string; (Optional) Label of the account.
+  * *tag* - string; (Optional) Tag for filtering accounts.
+  * *unlocked_balance* - unsigned int; Unlocked balance for the account.
+* *total_balance* - unsigned int; Total balance of the wallet (locked or unlocked).
```
Comment (moneromooo-monero) [inline _i18n/template/resources/developer-guides/wallet-rpc.md]: Incorrect, balance for the selected accounts (which may be the whole wallet in some cases)

File: _i18n/template/resources/developer-guides/wallet-rpc.md (hunk 9 above)
Comment (mattcode55) [inline _i18n/template/resources/developer-guides/wallet-rpc.md]: Should these be tagged as `(optional)`?

Comment (coneiric) [reply _i18n/template/resources/developer-guides/wallet-rpc.md]: Sure, I can update those as `(optional)`.

File: _i18n/template/resources/developer-guides/wallet-rpc.md (hunk 1 above)
Comment (coneiric) [reply _i18n/template/resources/developer-guides/wallet-rpc.md]: Thanks for catching this, will update with the correct information.

File: _i18n/template/resources/developer-guides/wallet-rpc.md (hunk 7 above)
Comment (coneiric) [reply _i18n/template/resources/developer-guides/wallet-rpc.md]: Ok, was a bit confused. Makes sense now, will update.

File: _i18n/template/resources/developer-guides/wallet-rpc.md (hunk 6 above)
Comment (coneiric) [reply _i18n/template/resources/developer-guides/wallet-rpc.md]: Ok, will update, don't know why I thought it was the keys.

File: _i18n/template/resources/developer-guides/wallet-rpc.md (hunk 8 above)
Comment (coneiric) [reply _i18n/template/resources/developer-guides/wallet-rpc.md]: This was to keep in line with the surrounding style (and got a bit lazy), I can add the description. Should I also add the other missing descriptions while I'm there?

File: _i18n/template/resources/developer-guides/wallet-rpc.md (hunk 9 above)
Comment (coneiric) [reply _i18n/template/resources/developer-guides/wallet-rpc.md]: Ok, will wait for your update to make changes here, if you like.

File: _i18n/template/resources/developer-guides/wallet-rpc.md (hunk 10 above)
Comment (coneiric) [reply _i18n/template/resources/developer-guides/wallet-rpc.md]: Right, oversight on my part, will update with the correct information.

**Comment by coneiric** [comment]
Thankyou for the thorough review @moneromooo-monero and @mattcode55 :+1:
Will make the requested changes, and update the PR.

**Comment by el00ruobuob** [comment]
Changes looks correct (but i'm not a dev). Could you copy the files in the other languages directory?
Then i could build and test before approving.

**Comment by el00ruobuob** [comment]
By the way, does this includes all the multisig-related commands?
I can see 12 times the multisig word here.
If so we should tell it solves #522

**Comment by coneiric** [comment]
> Changes looks correct (but i'm not a dev). Could you copy the files in the other languages directory?

@el00ruobuob I missed this part of the README, thank you for the reminder.

> By the way, does this includes all the multisig-related commands?

This PR does not include all the multisig-related commands. Those commands I did update for multisig, also needed subaddress updates.

I don't really have a lot of time at the moment to add the missing commands in this PR. I'll leave the few multisig-related updates for whomever adds the multisig docs (maybe me).

**Comment by el00ruobuob** [comment]
> I don't really have a lot of time at the moment to add the missing commands in this PR. I'll leave the few multisig-related updates for whomever adds the multisig docs (maybe me).

I'll do a full update of wallet RPC once this and #703 would be merged. I might ask you a few questions about them on #monero-dev at that time.

**Comment by el00ruobuob** [comment]
@coneiric could you remove the polish transalation from your PR? And also the template, to be compatible with the upcoming #744? And after that, i rebase would also be nice.
Sorry for asking that much, but i'm waiting for this to be merged on behalf of @rehrar to start working on the full refresh.

**Comment by coneiric** [comment]
@el00ruobuob I can make the changes you requested, and rebase the commits. So, now the workflow will be to implement the changes in the contributor's native language, then copy to other localizations?
Rebasing will squash the separated commits, but if that's fine with reviewers, I'll do it.

**Comment by el00ruobuob** [comment]
@coneiric the exact workflow must be pointed out by @erciccione in an update of readme.md soon.
At least, we now need to remove the template as it does not longer exist. And I believe we should include updates of all untranslated version when updating English, just to keep them up to date.
I'm working on a versioning solution to help us handle this part.
Regarding the squash I think it's not an issue to have it done before reviews. Anyway @luigi1111 will ask you to do it before it's merged.

**Comment by coneiric** [comment]
> Regarding the squash I think it's not an issue to have it done before reviews.

Alright, all requested changes are pushed. If reviewers still would like the changes in their split-up version, I created a backup branch. The changes to the language localizations are simple copies of those split-up changes.

File: _i18n/en/resources/developer-guides/wallet-rpc.md [hunk 11]
```diff
@@ -1,3 +1,4 @@
+{% include untranslated.html %}
```
Comment (erciccione) [inline _i18n/en/resources/developer-guides/wallet-rpc.md]: The English version doesn't need the 'untranslated' snippet

Comment (el00ruobuob) [reply _i18n/en/resources/developer-guides/wallet-rpc.md]: I missed it. Sorry @erciccione

**Comment by el00ruobuob** [comment]
Appart from the untranslated snippet mentioned by @erciccione it's ok on the following platforms:

  | Edge | IE 11 | Chrome | Firefox | Opera | Vivaldi
-- | -- | -- | -- | -- | -- | --
Windows 7 | - | OK | OK | OK | OK | OK
Windows 10 | OK | OK | OK | OK | OK | OK
Ubuntu 18.04 | - | - | OK | OK | OK | OK
CentOS 7 | - | - | - | OK | OK | OK

File: _i18n/en/resources/developer-guides/wallet-rpc.md (hunk 11 above)
Comment (coneiric) [reply _i18n/en/resources/developer-guides/wallet-rpc.md]: I'll fix it, thanks for the catch @erciccione.

**Comment by coneiric** [comment]
> No more 'Untranslated' snippet on English version, thanks.
Builds perfectly.

You're welcome, glad it's working now :+1:

**Comment by el00ruobuob** [comment]
ping @rehrar & @luigi1111 Could we merge this?

**Pull request merged**